src/components/LandingPageMiddleware.js

```diff
@@ -0,0 +1,35 @@
+import React from "react";
+import { Redirect } from "react-router-dom";
+import { DEFAULT } from "../constants";
+import { useModulesManager } from "../helpers/modules";
+import Contributions from "./generics/Contributions";
+
+/**
+ * @name LandingPageMiddleware
+ * @param {boolean} props.isAuthenticated
+ * @returns {React.Component}
+ *
+ * @description
+ * This component is a middleware that checks if the user is authenticated and if the landing page is enabled.
+ * If the user is not authenticated, it redirects to the login page.
+ * If the user is authenticated, it redirects to the home page.
+ * If the landing page is not enabled, it redirects to the home page (if the user is authenticated, otherwise to the login page).
+ * If the user is not authenticated and the landing page is enabled, it renders the landing page.
+ * To enable the landing page, set the "App.enableLandingPage" configuration to true and
+ * add the "core.LandingPage" contribution from any module.
+ */
+
+const LANDING_PAGE_CONTRIBUTION_KEY = "core.LandingPage";
```

Comment:
I know we discussed the landing page before but now that I'm looking at the code it may not be the best naming. It's right for the moldova use case but in here it's more like 'public page' not a 'landing page'. Can you consider changing the naming?

Reply:
Yes, that's a great idea, "public page" is the perfect fit here. It doesn't have to be a Landing Page; you can tailor it to whatever type of public page suits your needs, right. I'm changing the naming here.

Reply:
Perfect, thanks for the change.